docker-compose.yml

```diff
@@ -1,26 +1,32 @@
-version: '2.1'
+---
+version: '2.4'
 
 services:
-  cpo-case-payment-orders-api:
-    build:
-      context: .
-      args:
-        - http_proxy
-        - https_proxy
-        - no_proxy
-    image: hmctspublic.azurecr.io/spring-boot/template
+  case-payment-orders-api:
+    image: "hmctspublic.azurecr.io/cpo/case-payment-orders-api:latest"
+    container_name: case-payment-orders-api
     environment:
-      # these environment variables are used by java-logging library
-      - ROOT_APPENDER
-      - JSON_CONSOLE_PRETTY_PRINT
-      - ROOT_LOGGING_LEVEL
-      - REFORM_SERVICE_TYPE
-      - REFORM_SERVICE_NAME
-      - REFORM_TEAM
-      - REFORM_ENVIRONMENT
-      - LOGBACK_DATE_FORMAT
-      - LOGBACK_REQUIRE_THREAD
-      - LOGBACK_REQUIRE_ALERT_LEVEL=false
-      - LOGBACK_REQUIRE_ERROR_CODE=false
+      SPRING_DATASOURCE_URL: jdbc:postgresql://cpo-db:5432/cpo
```

Comment:
Should this correspond to the multiple variables used in the `src/main/resources/application.yaml`?

Reply:
yes, otherwise we need to define all 3 / 4 variables.

Reply:
NB: I am keen to but haven't tested your scripts (sorry, I currently cannot as I am mid-edit of my own work).

Is `SPRING_DATASOURCE_URL` a variable that would override [the equivalent in `application.yaml`](https://github.com/hmcts/cpo-case-payment-orders-api/blob/master/src/main/resources/application.yaml#L36)?  I had expected them to be specified separately as per [ccd-def-store](https://github.com/hmcts/ccd-definition-store-api/blob/master/docker-compose.yml#L10-L13).  

I am hoping our end solution allows FTAs to run against your `cpo-docker` but also debug version of CPO running at console or in IDE, I guess still using the database in your docker scripts.

Reply:
You can override any spring boot variable at any level. SPRING_DATASOURCE_URL is defined by concatenating multiple variables in our app. Either you can override individual ones or override parent variable itself (i.e  SPRING_DATASOURCE_URL). 
The app is up and able to run flyway scripts (that proves DB access).

Reply:
Thanks for the lesson on spring boot variables.  I am happy that your elements will all run together happily.